### PR TITLE
FLOW-1573: Update simple java mail and related code and tests

### PR DESCRIPTION
**Pull request by DaliborDell**

### CODE REVIEW

File: src/main/java/com/manywho/services/email/email/EmailManager.java [hunk 1]
```diff
@@ -22,9 +24,23 @@ public void send(ApplicationConfiguration configuration, Email email, boolean is
...
 
+        Mailer mailer = mailerFactory.createMailer(configuration);
+
         try {
-            mailerFactory.createMailer(configuration)
-                    .sendMail(email, sendAsynchronously);
+            AsyncResponse asyncTestConnection = mailer.testConnection(sendAsynchronously);
+            if (asyncTestConnection != null) {
+                asyncTestConnection.onException((e) -> {
+                    System.err.println("Connection error: " + e.getMessage());
```
Comment (josecollazzi) [inline src/main/java/com/manywho/services/email/email/EmailManager.java]: could you use the logger library that exist in the pom instead? it has many advantages.

File: src/main/java/com/manywho/services/email/email/EmailManager.java [hunk 2]
```diff
@@ -22,9 +24,23 @@ public void send(ApplicationConfiguration configuration, Email email, boolean is
...
+                });
+            }
+
+            AsyncResponse asyncSendMail = mailer.sendMail(email, sendAsynchronously);
+            if (asyncSendMail != null) {
+                asyncSendMail.onException((e) -> {
+                    System.err.println("Send mail error: " + e.getMessage());
```
Comment (josecollazzi) [inline src/main/java/com/manywho/services/email/email/EmailManager.java]: same here

**Review by josecollazzi — CHANGES_REQUESTED** [review]

File: src/main/java/com/manywho/services/email/email/EmailManager.java (hunk 1 above)
Comment (DaliborDell) [reply src/main/java/com/manywho/services/email/email/EmailManager.java]: Will update this to use logger :+1:

File: src/main/java/com/manywho/services/email/email/EmailManager.java (hunk 2 above)
Comment (DaliborDell) [reply src/main/java/com/manywho/services/email/email/EmailManager.java]: Will update to use the logger.

**Review by josecollazzi — APPROVED** [review]
lets remove the snapshot dependency before merge with master

**Review by josecollazzi — APPROVED** [review]

**Pull request merged**